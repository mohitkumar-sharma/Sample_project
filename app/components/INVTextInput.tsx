import React from 'react';
import { View, TextInput, StyleSheet, StyleProp, ViewStyle, TextInputProps } from 'react-native';
import INVDivider from './INVDivider';

export interface INVTextInputProps extends TextInputProps {
  /* as an object of style props for main content View element*/
  containerStyle?: StyleProp<ViewStyle>;
}

/**
 * This is the INVTextInput functional component works as a custom text input
 * @param {containerStyle, ...props} : containerStyle prop is defined in INVTextInputProps and ...props
 * related to props of TextInput
 */
const INVTextInput = ({ containerStyle, ...props }: INVTextInputProps): JSX.Element => {
  return (
    <View style={[styles.container, containerStyle]}>
      <TextInput style={styles.textInput} {...props} />
      <INVDivider />
    </View>
  );
};

const styles = StyleSheet.create({
  container: { marginHorizontal: 20 },
  textInput: { paddingHorizontal: 5, paddingVertical: 10 },
});

export default INVTextInput;
