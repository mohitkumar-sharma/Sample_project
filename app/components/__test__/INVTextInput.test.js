import React from 'react';
import { render } from '@testing-library/react-native';
import INVTextInput from '../INVTextInput';

describe('INVTextInput', () => {
  it('matches snapshot', () => {
    expect(render(<INVTextInput />)).toMatchSnapshot();
    expect(render(<INVTextInput containerStyle={{ margin: 20 }} />)).toMatchSnapshot();
  });
});
