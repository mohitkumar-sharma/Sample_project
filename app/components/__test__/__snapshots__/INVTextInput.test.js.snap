// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INVTextInput matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 20,
      },
      undefined,
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "paddingHorizontal": 5,
        "paddingVertical": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E8E8E8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`INVTextInput matches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "marginHorizontal": 20,
      },
      Object {
        "margin": 20,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "paddingHorizontal": 5,
        "paddingVertical": 10,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#E8E8E8",
          "height": 1,
        },
        undefined,
      ]
    }
  />
</View>
`;
